modules/javafx.graphics/src/test/java/test/javafx/scene/SceneTest.java

```diff
@@ -971,4 +973,52 @@ public void testAddNullPostLayoutPulseListener() {
         SceneShim.scenePulseListener_pulse(scene);
         assertEquals(1, counter.get());
     }
+
+    @Test public void testNoReferencesRemainToRemovedNodeAfterBeingClicked() {
+        StubToolkit toolkit = (StubToolkit) Toolkit.getToolkit();
+        MouseEventGenerator generator = new MouseEventGenerator();
+        TilePane pane = new TilePane();
+        VBox vbox = new VBox(pane);
+        Scene scene = new Scene(vbox, 300, 200);
+        WeakReference<TilePane> ref = new WeakReference<>(pane);
+
+        pane.setMinSize(1000, 1000);  // ensure mouse click will hit this node
+
+        stage.setScene(scene);
+
+        // Press mouse on TilePane so it gets picked up as a potential node to drag:
+        SceneHelper.processMouseEvent(
+            scene,
+            generator.generateMouseEvent(MouseEvent.MOUSE_PRESSED, 100, 100)
+        );
+
+        // Release mouse on TilePane to trigger clean up code as there will be no dragging:
+        SceneHelper.processMouseEvent(
+            scene,
+            generator.generateMouseEvent(MouseEvent.MOUSE_RELEASED, 100, 100)
+        );
+
+        // Remove TilePane, and replace with something else:
+        vbox.getChildren().setAll(new StackPane());
+
+        // Generate a MOUSE_EXITED event for the removed node and a pulse as otherwise many unrelated Scene references
+        // hang around to the removed node:
+        SceneHelper.processMouseEvent(
+                scene,
+                new MouseEvent(
+                        MouseEvent.MOUSE_EXITED, 100, 100, 100, 100, MouseButton.NONE, 0, false, false, false,
+                        false, false, false, false, false, false, true, null
+                )
+        );
+
+        toolkit.firePulse();
+
+        // Clear our own reference and call GC to see if the TilePane is now not referenced anywhere:
+        pane = null;
+
+        System.gc();
```

Comment:
It is not sufficient to call `System.gc()` only once, since it will make for a fragile test. I recommend using the `JMemoryBuddy::assertCollectable` for this, which is what we are using for new tests.

Reply:
This tool is in javafx.base's test folder, and it is not allowing me to reference it.  I'm not quite sure how to change the project to allow this.